src/libraries/Microsoft.Extensions.Caching.Memory/src/Microsoft.Extensions.Caching.Memory.csproj

```diff
@@ -14,4 +14,8 @@
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Primitives\src\Microsoft.Extensions.Primitives.csproj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
+  </ItemGroup>
```

Comment:
Backporting a fix that has new dependencies seems problematic.

Reply:
Should I create a new private `struct` for this data instead of using a ValueTuple?

Reply:
@ericstj @davidfowl - any thoughts here? If we decide to backport this to `6.0.x`, would this new dependency on the `System.ValueTuple` package be a problem? Nothing in `MS.Ext.Caching.Memory` brings in this package today on netfx.

I can easily switch to not use value tuples, but it seems like a pain going forward.

One option could be use value tuples in `7.0` going forward, and just use a custom `struct` in the `6.0` version.

Reply:
I would just stick to a small internal struct for now. It's much less intrusive and doesn't require an ifdef. We should then decide when we rip the netstandard 2.0 cord. (.NET 7 or 8?)

Reply:
> doesn't require an ifdef

The plan I'm proposing doesn't require an ifdef (beyond this PackageReference). The idea is:

For the `7.0.0`+ versions of this package (in `main`) we use ValueTuple like I am here. Then it is available in the future if we want to use it in other places. If servicing `6.0` wasn't a consideration, this is the plan I would take without a 2nd thought.

For the `6.0.1` version of this package (in `release/6.0`) change the ValueTuple uses in this PR to a small struct and remove the dependency on System.ValueTuple. That way we don't add the new dependency in a service release.

Reply:
The `ValueTuple` dependency can be a real PITA for users who are using a very old VS that generates invalid binding redirects for .NET Framework projects: https://github.com/dotnet/BenchmarkDotNet/issues/1687#issuecomment-818622556

I doubt that they use latest MemoryCache version and I think that we should just stop targeting .NET Standard 2.0. But we need to verify that with some data.